test/test-lib/api-helpers.ts

```diff
@@ -81,33 +82,45 @@ export const addImageToRelease = async (
 		.expect(201);
 };
 
-export const expectResourceToMatch = async (
-	pineUser: PineTest,
+export const expectResourceToMatch = async <T = AnyObject>(
+	pineUser: PineTest | PinejsClient | { token: string } | string,
 	resource: string,
 	id: number | AnyObject,
 	expectations: Dictionary<
 		| null
 		| string
 		| number
 		| boolean
-		| ((chaiPropertyAssetion: Chai.Assertion) => void)
+		| object
+		| ((chaiPropertyAssertion: Chai.Assertion) => void)
 	>,
-) => {
-	const { body: result } = await pineUser
-		.get({
-			resource,
-			id,
-			options: {
-				$select: Object.keys(expectations),
-			},
-		})
-		.expect(200);
-	for (const [key, valueOrAssertion] of Object.entries(expectations)) {
-		if (typeof valueOrAssertion === 'function') {
-			valueOrAssertion(expect(result).to.have.property(key));
+): Promise<T> => {
+	if (typeof pineUser === 'string' || 'token' in pineUser) {
+		pineUser = pineTest.clone({ passthrough: { user: pineUser } });
+	}
+
+	const pineClientRequest = pineUser.get({
+		resource,
+		id,
+		options: {
+			$select: Object.keys(expectations),
+		},
+	});
+	const result = (
+		'expect' in pineClientRequest
```

Comment:
Uncommon, but do I understand correctly that the pineClientRequest has a key `expect` ?

Reply:
So at this point pineUser is either a pinejs-client-supertest or a pinejs-client-request (the one the pinejs exposes) instance, and the badly named `pineClientRequest` variable here is a request object (promise). So if it's from `pinejs-client-supertest`, then it will also have a `.expect()` (since it's a supertest + pinejs-client hybrid). So here we are doing a feature check (rather than instance check) and if `.expect` is available, we use it. (let me rename the var...)